### PR TITLE
certbot: bump python resources to latest version

### PR DESCRIPTION
**Pull request by alex**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by alex** [comment]
Hmm, so the way this is failing is exactly what I'd expect if you tried to manually install packages with setuptools, and without using `pip`. The part that confuses me is that it's only showing up on this package, and not _every_ python package.

**Comment by alebcay** [comment]
The cause for failure is that the `cryptography` package is unable to build. This is happening because we're not installing the `setuptools-rust` package, which is used at build-time for `cryptography` (I see that `rust` has been added as build-time dependency but the `setuptools-rust` package is the interface that allows for the Rust bits of `cryptography` to get built in the first place). Like you mentioned, installing via `pip` would not run into this issue because `setuptools-rust` gets pulled in as a build-time tool dependency in accordance with PEP 517.

tl;dr: The issue is happening because `cryptography` in particular has PEP 517 build-time tool dependencies that setuptools cannot satisfy.

Several general approaches for the issue:
- Add `setuptools-rust` as a resource and handle it and `cryptography` separately in the installation process
- Switch from a setuptools-based installation process to a pip-based one

**Comment by alex** [comment]
So, the piece I don't understand is why none of what you just said applies to other packages using cryptography 3.4+ (FWIW, I'm one of the cryptography developers, that's my interest here).

Here's an example: https://github.com/Homebrew/homebrew-core/blob/master/Formula/ykman.rb#L40 no mention of setuptools-rust here.

I think the issue here must be that `Language::Python::setup_install_args` uses `setup.py` and not `pip`. So I guess I need to go learn about that.

**Comment by alex** [comment]
I think the right solution here is to convert this to use `virtualenv_create`, however before I refactor to do that I want to ensure I understand if there's a reason it wasn't written that way before. I don't suppose anyone knows?

**Comment by alebcay** [comment]
> Here's an example: https://github.com/Homebrew/homebrew-core/blob/master/Formula/ykman.rb#L40 no mention of setuptools-rust here.

Good catch. `ykman` (and many others that depend on `cryptography`) use `virtualenv_install_with_resources`, which use a [`pip`-based installation flow](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/language/python.rb#L288-L290):
```
pip install -v --no-deps --no-binary :all: --ignore-installed <target>
```

What this means is that `setuptools-rust` is pulled in transparently by `pip` (`--no-deps` has no power there when it comes to build dependencies). This behavior is only really something we've caught on to recently (we don't really follow the Python packaging world too closely, especially if the new PEP 517 builds "just work" so we didn't really notice the change).

It's something we're trying to avoid/work around. From a package manager perspective, having (non-pinned) build-time dependencies that get pulled in transiently isn't too great.

> I want to ensure I understand if there's a reason it wasn't written that way before. I don't suppose anyone knows?

I don't think there's really any particular reason. It may just be a style difference from back in the days when `python setup.py install [...]` was more or less synonymous with `pip install --no-binary :all: --no-deps [...]`, although this is not really the case anymore for a number of packages.

> I'm one of the cryptography developers, that's my interest here

I knew your name looked familiar :)
